Replace type in CapabilityData::EccCurves

ionut-arm opened the pull request:
Replacing the FFI type with its abstraction for the contents of ECC
Curve capabilities.
The CapabilityData tests are also enhanced. A few of the variants are
renamed to follow the proper capitalization rules.

Signed-off-by: Ionut Mihalcea <ionut.mihalcea@arm.com>

Superhepper commented:
Do you think it would make any sense to wrap this up in a list type and then maybe add a 'contains' function in order to make it easier to determine if a certain curve is supported?

ionut-arm commented:
> Do you think it would make any sense to wrap this up in a list type and then maybe add a 'contains' function in order to make it easier to determine if a certain curve is supported?

Hmm, `Vec` already has a [`contains`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.contains) method that does exactly that, and I think `EccCurveIdentifier` has the right traits (just `PartialEq`, actually)

ionut-arm commented:
Oh, actually, given that those are just enum variants, I think the best option would actually be `HashSet`, to ensure no duplicates exist. Will change that.

Superhepper commented:
The reason I like wrapping it up is so that we can if needed change the underlaying container without breaking the api.

ionut-arm commented:
> The reason I like wrapping it up is so that we can if needed change the underlaying container without breaking the api.

Ok, done, I've added an `EccCurveList` type. I noticed the `Handles` variant in `CapabilityData` had the same issue, so I changed that one too. I've not actually added a `contains` method, since that one is inherited from the `Vec`, but we can implement one in the future if we need to change it.

Superhepper reviewed (approved):
Nice! This will make things nice and consistent.

wiktor-k reviewed (approved):
This looks very nice, thanks! :+1:
